web/satellite/src/components/project/paymentMethods/DeletePaymentMethodDialog.vue

```diff
@@ -24,54 +24,46 @@
     } from '@/utils/constants/actionNames';
 
     @Component({
-        props: {
-            paymentMethodID: {
-                type: String,
-                default: ''
-            },
-        },
-        methods: {
-            onCancelClick: function () {
-                this.$store.dispatch(APP_STATE_ACTIONS.CLOSE_POPUPS);
-            },
-            onDeleteClick: async function () {
-                const response = await this.$store.dispatch(PROJECT_PAYMENT_METHODS_ACTIONS.DELETE, this.$props.paymentMethodID);
-                if (!response.isSuccess) {
-                    this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, response.errorMessage);
-                }
-
-                const paymentMethodsResponse = await this.$store.dispatch(PROJECT_PAYMENT_METHODS_ACTIONS.FETCH);
-                if (!paymentMethodsResponse.isSuccess) {
-                    this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, 'Unable to fetch payment methods: ' + paymentMethodsResponse.errorMessage);
-                }
-
-                this.$store.dispatch(NOTIFICATION_ACTIONS.SUCCESS, 'Successfully delete payment method');
-            }
-        },
         components: {
             Button,
         }
     })
     export default class DeletePaymentMethodDialog extends Vue {
+        @Prop({default: ''})
+        private readonly paymentMethodID: string;
+
+        public onCancelClick(): void {
+            this.$store.dispatch(APP_STATE_ACTIONS.CLOSE_POPUPS);
+        }
+
+        public async onDeleteClick(): Promise<void> {
+            const response = await this.$store.dispatch(PROJECT_PAYMENT_METHODS_ACTIONS.DELETE, this.paymentMethodID);
+            if (!response.isSuccess) {
+                this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, response.errorMessage);
+            }
+
+            const paymentMethodsResponse = await this.$store.dispatch(PROJECT_PAYMENT_METHODS_ACTIONS.FETCH);
+            if (!paymentMethodsResponse.isSuccess) {
+                this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, 'Unable to fetch payment methods: ' + paymentMethodsResponse.errorMessage);
```

Comment:
lets use template string here because it harder to make type like this 'Unable to fetch payment methods: ' + +paymentMethodsResponse.errorMessage
`Unable to fetch payment methods: ${paymentMethodsResponse.errorMessage}`